src/ca.c

```diff
@@ -2885,17 +2887,13 @@ char *get_ca_pin(int i) {
 void set_ca_pin(int i, char *pin) {
     if (!ca_devices[i])
         ca_devices[i] = alloc_ca_device();
-    if (!ca_devices[i])
-        return;
```

Comment:
I dropped these early returns because there was at least one place where this was _not_ done, so I figured it's pointless since we'll crash anyway.

Reply:
It is not really pointless... my tests on no free memory shows that minisatip still works fairly good.

Reply:
I can drop the change and instead add the check where it's missing?

Reply:
Sounds good